.github/workflows/ci.yml

```diff
@@ -0,0 +1,67 @@
+# This workflow performs tests in JavaScript.
+name: ESDK JavaScript CI Tests 
+
+on: [pull_request, workflow_call]
+
+jobs:
+  CI:
+    strategy:
+      matrix:
+          node: [16, 18.3.0, 20.4.0]
+      fail-fast: false
+    runs-on: codebuild-AWS-ESDK-JS-Release-${{ github.run_id }}-${{ github.run_attempt }}-ubuntu-5.0-large
+    permissions:
+      id-token: write
+      contents: read
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true 
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{matrix.node}}
+      - name: Configure AWS Credentials for Tests
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::370957321024:role/GitHub-CI-MPL-Dafny-Role-us-west-2
+          role-session-name: JavaScriptTests 
+      - name: Test Node ${{matrix.node}}
+        env: 
+          NODE_OPTIONS: "--max-old-space-size=4096"
+        run: |
+          npm ci --unsafe-perm
+          npm run build
+          npm run coverage-node
+      - name: Test compliance
+        env: 
+          NODE_OPTIONS: "--max-old-space-size=4096"
+        run: |
+          npm ci --unsafe-perm
```

Comment:
Suggestion: Running `npm ci` once SHOULD be sufficient.
```suggestion
```

Reply:
```suggestion
          npm ci
```